apis/v1beta1/config.go

```diff
@@ -530,6 +615,48 @@ type Telemetry struct {
 	Resource map[string]*string `json:"resource,omitempty" yaml:"resource,omitempty"`
 }
 
+func (t *Telemetry) toMap() (map[string]interface{}, error) {
+	data, err := json.Marshal(t)
+	if err != nil {
+		return nil, err
+	}
+	var result map[string]interface{}
+	if err := json.Unmarshal(data, &result); err != nil {
+		return nil, err
+	}
+
+	normalizeConfig(result)
+
+	return result, nil
+}
+
+func normalizeConfig(m map[string]interface{}) {
+	for k, v := range m {
+		switch val := v.(type) {
+		case nil:
+			// We remove those fields which value is nil. This prevents issues when
+			// unmarshalling the config in the collector
+			delete(m, k)
+		case map[string]interface{}:
+			normalizeConfig(val)
+		case []interface{}:
+			for i, item := range val {
+				if item == nil {
+					val[i] = map[string]interface{}{}
+				} else if sub, ok := item.(map[string]interface{}); ok {
+					normalizeConfig(sub)
+				}
+			}
+		case float64:
+			// All numbers (even int32, int64, etc.) are parsed as float64 by default
+			// We need to convert them to the correct type
+			if k == "port" {
+				m[k] = int32(val)
+			}
+		}
```

Comment:
Should have a default case here as well.

Reply:
Nothing. I think it's a good practice to always specify what happens if none of the cases match, but I don't care that much if you don't want to change it.

Reply:
Didn't know about that. Thanks!